tests/Unit/ConfigurationTest.php

```diff
@@ -23,6 +23,42 @@ public function testValidConfig(array $inputConfig, array $expectedConfig): void
         $this->assertProcessedConfigurationEquals($inputConfig, $expectedConfig);
     }
 
+    /**
+     * @param mixed $value
+     *
+     * @dataProvider dataTestSettingsDynamicChecker
+     */
+    public function testSettingsDynamicChecker($value, bool $passed): void
+    {
+        if (!$passed) {
+            $this->assertConfigurationIsInvalid([
+                'meilisearch' => [
+                    'indices' => [
+                        [
+                            'name' => 'items',
+                            'class' => 'App\Entity\Post',
+                            'settings' => $value,
+                        ],
+                    ],
+                ],
+            ], 'Settings must be an array.');
+
+            return;
+        }
+
+        $this->assertConfigurationIsValid([
```

Comment:
better would be to split into separate tests than mixing in a single one valid and invalid values